source/windows/bcrypt_hash.c

```diff
@@ -86,17 +108,38 @@ static void s_load_md5_alg_handle(void *user_data) {
         s_md5_alg, BCRYPT_OBJECT_LENGTH, (PBYTE)&s_md5_obj_len, sizeof(s_md5_obj_len), &result_length, 0);
 }
 
-struct aws_hash *aws_sha256_default_new(struct aws_allocator *allocator) {
-    aws_thread_call_once(&s_sha256_once, s_load_sha256_alg_handle, NULL);
+struct aws_hash *aws_sha512_default_new(struct aws_allocator *allocator) {
+    aws_thread_call_once(&s_sha512_once, s_load_sha512_alg_handle, NULL);
 
     struct bcrypt_hash_handle *bcrypt_hash = NULL;
     uint8_t *hash_obj = NULL;
-    aws_mem_acquire_many(allocator, 2, &bcrypt_hash, sizeof(struct bcrypt_hash_handle), &hash_obj, s_sha256_obj_len);
+    aws_mem_acquire_many(allocator, 2, &bcrypt_hash, sizeof(struct bcrypt_hash_handle), &hash_obj, s_sha512_obj_len);
 
-    if (!bcrypt_hash) {
+    AWS_ZERO_STRUCT(*bcrypt_hash);
+    bcrypt_hash->hash.allocator = allocator;
+    bcrypt_hash->hash.vtable = &s_sha512_vtable;
+    bcrypt_hash->hash.impl = bcrypt_hash;
+    bcrypt_hash->hash.digest_size = AWS_SHA512_LEN;
+    bcrypt_hash->hash.good = true;
+    bcrypt_hash->hash_obj = hash_obj;
+    NTSTATUS status = BCryptCreateHash(
+        s_sha512_alg, &bcrypt_hash->hash_handle, bcrypt_hash->hash_obj, (ULONG)s_sha512_obj_len, NULL, 0, 0);
+
+    if (((NTSTATUS)status) < 0) {
+        aws_mem_release(allocator, bcrypt_hash);
         return NULL;
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/api/bcrypt/nf-bcrypt-bcryptcreatehash
This function seems to have some error code, if we can convert them to our error code and then return null, it will be more consistent with the other code base.

But, it seems to be like this for c-call all the time. Maybe we can add a todo or something for better error handling in c-cal.